### ISSUE TITLE
Alien Breed 3D II vsync issues

### ISSUE THREAD

**Issue by ZyberSE** [BlitterStudio/amiberry#1047]
**Describe the bug**
Work is being done on AB3DII and I've been testing the new builds for a while, both on Pimiga 3 (with manually updated/compiled Amiberry from this repo) and WinUAE on my Windows laptop.

On WinUAE it runs flawlessly, but in Amiberry I get this wired, vsync looking graphics issues when moving around. The entire screen stutters so it's difficult playing. I've tried both overclocked and stock, but with the same result.

Video of the bug:
https://imgur.com/a/zHaV6dF

**To Reproduce**
1) Install AB3DII as normall
2) Download this file: https://eab.abime.net/attachment.php?attachmentid=77607&d=1672522644
3) Extract it into the AB3DII folder
4) Start TKG_new
5) Enable the mouse (press m) and move around, you'll see the issue

Here you can find all discussions about the new builds:
https://eab.abime.net/showthread.php?t=111090&goto=newpost

Here is the Github repo:
https://github.com/mheyer32/alienbreed3d2

**Expected behavior**
Run smooth. On an overclocked RPi 400 I see between 25-50 fps.

**Screenshots**
https://imgur.com/a/zHaV6dF

**Desktop (please complete the following information):**
- Pimiga 3
- Amiberry 5.5.1

**Comment by midwan** [comment]
@ZyberSE 
Hm, interesting glitch. Have you tried disabling the Multithreaded Drawing option, that Amiberry has? That can cause similar kind of artifacts, that I've noticed in some demos.

You can find the option in the Chipset panel, but you cannot change it on-the-fly. Stop emulation, change it, then start it again (or load your config, change it, then start emulation).

**Comment by ZyberSE** [comment]
I've just tested and this link works for me, see if you can download it:
https://eab.abime.net/attachment.php?attachmentid=77607&d=1672522644

I checked my settings and Multithreaded Drawing was disabled so I enabled it just to see what happened, gfx was the same but the whole game felt a bit faster. Didn't do any back-to-back testing so could be placebo.

**Comment by midwan** [comment]
@ZyberSE 
Sorry, that link still doesn't work for me (same error). 
The multithreaded option does offer a performance benefit, which is why it's there (and it doesn't exist in WinUAE). But since it _can_ have glitches similar to vsync not working (not as extreme as shown in your video however), it's optional.

Perhaps it's another setting then. Does it do the same with and without JIT enabled? Have you changed any other settings to see if they affect it?

If I can get a binary I can run, I could test it here and perhaps figure out what's causing it. I also have a newer, more compatible version I can test with.

**Comment by ZyberSE** [comment]
I've played around a bit yes, never managed to get any real improvements though. Best I managed was to switch between pal/ntsc but that introduced some other issues instead.
With JIT disabled the game runs so slow you can't really make anything of it.

If you use Discord I can send you the file? Just drop me a message ZyBeR#8378

**Comment by ZyberSE** [comment]
@midwan I wounded if you need to have the "zone" rights to download files? You can add that in your account settings somewhere, can't remember exactly where.

**Comment by nemo93** [comment]
The `.lha` version (`AlienBreed3D2_v1.1_AGA_0280.lha`) does work fine here on a RPi4/Retropie default setup. I can't download the files from the provided links either.

For the record, the default settings applied for that game are: CPU max speed, no JIT, **Fast Copper yes** (one of the few).

**Comment by midwan** [comment]
@ZyberSE I do have access to the Zone, so it's not that I'm afraid. It's not there from what I can see, right?
Do you have another way of sharing the binaries?

**Comment by ZyberSE** [comment]
@midwan  @nemo93 Try downloading it from here: https://wormhole.app/nX7b7#ecx4RILTtgacMRyKcOgZFQ
It's only available for 24h from now.

**Comment by midwan** [comment]
@ZyberSE Thank you, that worked. And I eventually found the attachments in the forum thread as well, but good to know I have the latest version. :)

I'll run some tests here later today and get back to you.

**Comment by ZyberSE** [comment]
Cool @midwan , let me know if I can be of any assistance. I've tested without JIT and with Fast Copper but the problems persists, without JIT it's too slow to to be usable though.

**Comment by midwan** [comment]
@ZyberSE 
I did a quick test on my Pi400, running a clone of my real A4000's AmigaOS 3.2.1 installation.
I don't see the problem you posted above, it seems to be working normally here. Perhaps some hack/patch that Pimiga is running on startup is messing things up for you?

**Comment by ZyberSE** [comment]
Interesting @midwan, mind to share your Amiberry config so I can try it?

**Comment by midwan** [comment]
@ZyberSE 
I'm attaching my `amiberry.conf` and `A4000.uae` configs that I used:
[A4000.zip](https://github.com/BlitterStudio/amiberry/files/10480102/A4000.zip)

**Comment by ZyberSE** [comment]
I've done some initial testing, only difference in amiberry.conf "default_line_mode=1", I had it to 0. Not sure what that setting does though? Made no difference.

Also you have cpu idle set in your A4000.uae, I tried it but no improvement. What does that thing do?

I'm thinking it might have something todo with the roms then?
If you have a spare sd-card, would you have time to test Pimiga 3 with your roms?

**Comment by midwan** [comment]
@ZyberSE 
Line mode corresponds to the options in Display (Single, Double, etc.)

Chances are the problem is due to something running inside AmigaOS in Pimiga, not the emulator itself.
I can test it there also a bit later...

**Comment by ZyberSE** [comment]
I just did a test with A1200 3.2.1 roms but still had the same issue.

**Comment by midwan** [comment]
@ZyberSE I can't see how anything in the ROMs would affect this - try with a clean AmigaOS installation instead, and check there.

I tested it in the Pimiga3 installation here, and I can see it also. But not in my own AmigaOS 3.2.1 installation, which means it's something within the OS that's causing it perhaps. There are a bunch of patches/hacks starting up in Pimiga, so it could be any one of them.

That's why I think the best test would be with a clean system (3.1, 3.9, 3.2.1, doesn't matter).

**Comment by midwan** [comment]
@ZyberSE 
Any luck with this? Did you get a chance to test it on a clean system?

**Comment by ZyberSE** [comment]
Yeah I've done some testing. Copied my system from WinUAE to the Pi400 with Amiberry and ran it, same roms and same Amiberry settings. It's better, higher fps but still the vsync (I think) issues.
See videos here: https://imgur.com/a/kzRabdp

**Comment by midwan** [comment]
@ZyberSE 
Not sure if we can do more about this here, considering:
- We established that it happens in Pimiga, but not in other / clean systems
- Your videos show a drop of FPS, so in that case it's "normal". The display updates 50/60 times per second, but if your framerate drops below that, you will notice some hiccups in what is drawn.
- I couldn't recreate this in a clean system here, while my Pi400 maintained a steady 50 FPS in the game.

Therefore I'm closing this for now - feel free to add more details if you have them.
